WireDomain/Sources/WireDomain/Synchronization/NewIncrementalSync.swift

```diff
@@ -0,0 +1,214 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Combine
+import Foundation
+import WireAPI
+import WireLogging
+
+public protocol LiveSyncDelegate {
+    func didFinishSync(sync: NewIncrementalSync)
+    func didMissedEvents(sync: NewIncrementalSync) async
+    func didFail(sync: NewIncrementalSync, error: any Error)
+}
+
+/// IncrementalSync using new backend API async stream notifications
+public struct NewIncrementalSync: LiveSyncProtocol {
+
+    private let selfClientID: String
+    private let pushChannelAPI: any NewPushChannelAPI
+    private let decryptor: any UpdateEventDecryptorProtocol
+    private let store: any UpdateEventsLocalStoreProtocol
+    private let processor: any UpdateEventProcessorProtocol
+    private let databaseSaver: any DatabaseSaverProtocol
+    private let syncStateSubject: CurrentValueSubject<SyncState, Never>
+    private let logger = WireLogger.sync
+    var delegate: (any LiveSyncDelegate)?
+
+    public init(
+        selfClientID: String,
+        pushChannelAPI: any NewPushChannelAPI,
+        decryptor: any UpdateEventDecryptorProtocol,
+        store: any UpdateEventsLocalStoreProtocol,
+        processor: any UpdateEventProcessorProtocol,
+        databaseSaver: any DatabaseSaverProtocol,
+        syncStateSubject: CurrentValueSubject<SyncState, Never>
+    ) {
+        self.selfClientID = selfClientID
+        self.pushChannelAPI = pushChannelAPI
+        self.decryptor = decryptor
+        self.store = store
+        self.processor = processor
+        self.databaseSaver = databaseSaver
+        self.syncStateSubject = syncStateSubject
+    }
+
+    public func perform(acknowledgeFullSync: Bool) async throws -> IncrementalSync.Token {
+        logger.debug("performing live sync v3")
+        let pushChannel = try await pushChannelAPI.createPushChannel(clientID: selfClientID)
+
+        if acknowledgeFullSync {
+            try await pushChannel.ackFullSync()
+        }
+
+        logger.debug("opening new push channel v3")
+        syncStateSubject.send(.incrementalSyncing(.openPushChannel))
+        let liveEventStream = try await pushChannel.open()
+
+        let task: Task<Void, Error> = Task { @Sendable [
+            logger,
+            decryptor,
+            store,
+            processor,
+            databaseSaver,
+            pushChannel,
+            delegate,
+            syncStateSubject
+        ] in
```

Comment:
some Sendable warnings here, not sure how to resolve them, will cleanup in a following PR